modules/renter/upload.go

```diff
@@ -74,6 +70,15 @@ func (r *Renter) Upload(up modules.FileUploadParams) error {
 			up.PieceSize = smallPieceSize
 		}
 	}
+	// calculate end height
+	// TODO: using 0 as a special value is kind of hacky. Should probably have
+	// an explicit "renew" field.
+	var endHeight types.BlockHeight
+	if up.Duration == 0 {
+		endHeight = 0
+	} else {
+		endHeight = r.cs.Height() + up.Duration
+	}
```

Comment:
`duration=0` is used to indicate that the file should be indefinitely renewed. Seems preferable to make the call accept either `duration=n` or `renew=true` (or maybe both?), although this complicates the API somewhat.
I would actually like to do this in a separate PR, since it touches the API. It seems significant enough for it to stand on its own.